server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2076,6 +2076,13 @@ public IndexMetadata build() {
                 );
             }
 
+            var aliasesMap = aliases.build();
+            aliasesMap.forEach((key, value) -> {
```

Comment:
It seems that only check the value, not the key, so can just iterate over a `aliases.value()` view?
```java
 for (AliasMetadata value : aliasesMap.values()) {
                if (value.alias().equals(index)) {
                    throw new IllegalArgumentException("alias name [" + index + "] and index name may not be the same");
                }
            }
```